src/Spago/Build.hs

```diff
@@ -200,33 +202,117 @@ run
 run maybeModuleName buildOpts extraArgs = do
   Config.Config { alternateBackend } <- view (the @Config)
   let moduleName = fromMaybe (ModuleName "Main") maybeModuleName
-  runBackend alternateBackend moduleName Nothing "Running failed; " buildOpts extraArgs
+  runBackend alternateBackend Nothing moduleName Nothing "Running failed; " buildOpts extraArgs
 
 
+-- | Run the select module as a script: init, compile, and run the provided module
+script
+  :: (HasEnv env)
+  => Text
+  -> Maybe Text
+  -> [PackageName]
+  -> RIO env ()
+script modulePath tag packageDeps = do
+  logDebug "Running `spago script`"
+  absoluteModulePath <- fmap Text.pack (makeAbsolute (Text.unpack modulePath))
+  moduleContents <- readTextFile (pathFromText absoluteModulePath)
+
+  moduleName <- case parseModuleName moduleContents of
+    Nothing -> die [ "Failed to extract module name from source file" ]
+    Just modName -> pure modName
+
+  GlobalCache cacheDir _ <- view (the @GlobalCache)
+  currentDir <- Turtle.pwd
+  Temp.withTempDirectory cacheDir "spago-script-tmp" $ \dir -> do
+    let tmpDir = Text.pack dir
+    Turtle.cd (Turtle.fromText tmpDir)
+
+    config@Config { packageSet = PackageSet{..}, ..}
+      <- Packages.initProject NoForce Dhall.WithComments tag
+
+    let newConfig :: Config
+        newConfig = config
+          { Config.dependencies = dependencies <> packageDeps
+          , Config.configSourcePaths = []
+          }
+
+    Run.withInstallEnv' (Just newConfig) installAction
+
+    let runDirs :: RunDirectories
+        runDirs = RunDirectories tmpDir currentDir
+
+    Run.withBuildEnv' (Just newConfig) NoPsa (runAction absoluteModulePath moduleName runDirs)
+  where
+    installAction = do
+      deps <- Packages.getProjectDeps
+      Fetch.fetchPackages deps
+
+    runAction absoluteModulePath moduleName dirs = do
+      let
+        buildOpts = BuildOptions
+          { shouldWatch = BuildOnce
+          , shouldClear = NoClear
+          , allowIgnored = DoAllowIgnored
+          , sourcePaths = [ SourcePath absoluteModulePath ]
+          , withSourceMap = WithoutSrcMap
+          , noInstall = NoInstall
+          , pursArgs = []
+          , depsOnly = AllSources
+          , beforeCommands = []
+          , thenCommands = []
+          , elseCommands = []
+          }
```

Comment:
@f-f we could use a careful look here -- we weren't sure what to set for some of these, specifically the `pursArgs` field.

Reply:
I think `clear`, `purs-args`, `before`, `then`, and `else` make sense.

I tried to see what it would be like to enable `Watch`, and I hit problems where I think the temp directory was deleted before a second build got kicked off, and `purs compile` couldn't find the dependencies anymore. I'm not sure why...

Reply:
@colinwahl no worries, let's give up on the `watch` for now - it's more important to ship this first, and we can revisit the `watch` later